saleor/warehouse/management.py

```diff
@@ -178,8 +199,18 @@ def deallocate_stock(order_lines_data: Iterable["OrderLineData"]):
     if not_dellocated_lines:
         raise AllocationError(not_dellocated_lines)
 
+    available_stock_before = [
+        (allocation, allocation.stock.available_quantity())
```

Comment:
Multiple database queries. 
We fetch stock and in the next query fetch `available_quantity` for this query. 
We should fetch it like `Allocations.object.filter(stock_id=allocation.stock_id).aggregate(Sum("quantity_allocated"))`